packages/amplify-cli-core/src/cfnUtilities.ts

```diff
@@ -1,12 +1,26 @@
-import * as yaml from 'js-yaml';
 import { Template } from 'cloudform-types';
-import { JSONUtilities } from './jsonUtilities';
 import * as fs from 'fs-extra';
+import * as yaml from 'js-yaml';
 import * as path from 'path';
+import { JSONUtilities } from './jsonUtilities';
+
+export async function readCFNTemplate(filePath: string): Promise<{ templateFormat: CFNTemplateFormat; cfnTemplate: Template }>;
+export async function readCFNTemplate(
+  filePath: string,
+  options: { throwIfNotExist: boolean },
+): Promise<{ templateFormat: CFNTemplateFormat; cfnTemplate: Template } | undefined>;
+
+export async function readCFNTemplate(filePath: string, options: { throwIfNotExist: boolean } = { throwIfNotExist: true }) {
```

Comment:
A pattern I like to follow for the options with defaults is:
```suggestion
const defaultReadCFNTemplateOptions = {
  throwIfNotExists: true
}

export async function readCFNTemplate(filePath: string, options: Partial<typeof defaultReadCFNTemplateOptions> = defaultReadCFNTemplateOptions) {
options = {...defaultReadCFNTemplateOptions, ...options};
```

Alternatively, if this is going to be the only option that could potentially make the function return undefined, maybe it would be better to simply define another function:
```
export async function readCFNTemplateSafe(filePath: string): Promise<{ templateFormat: CFNTemplateFormat; cfnTemplate: Template } | undefined>
```

I'll leave it up to you
```